### ISSUE TITLE
Not able to create announcements without refreshing page

### ISSUE THREAD

**Issue by kcalvelli** [nextcloud/announcementcenter#267]
### Steps to reproduce
1 - Load announcementcenter
2 - Fill out subject and body fields
3 - Click announce
4 - Nothing happens
5 - Refresh Page
6 - Click announce 
7 - Announcement posted

### Expected behaviour
1 - Should see group field
2 - Clicking announce should create announcement
3 - Should not need to refresh page

### Actual behaviour
1 - Group field missing 
2 - Buttons unresponsive
3 - Refresh page and group field appears, buttons become responsive

### Server configuration

**Operating system**: Linux 5.9.16-100.fc32.x86_64 #1 SMP Mon Dec 21 14:10:00 UTC 2020 x86_64

**Web server:** nginx/1.18.0 (fpm-fcgi)

**Database:** mysql 10.5.8

**PHP version:** 7.3.25
Modules loaded: Core, date, libxml, pcre, zlib, filter, hash, readline, Reflection, SPL, session, cgi-fcgi, bcmath, bz2, ctype, curl, dom, fileinfo, ftp, gd, gmp, iconv, intl, json, ldap, mbstring, openssl, pcntl, PDO, pgsql, posix, standard, SimpleXML, smbclient, sodium, sqlite3, xml, xmlwriter, zip, exif, imap, mysqlnd, pdo_pgsql, pdo_sqlite, Phar, xmlreader, pdo_mysql, apcu, igbinary, redis, memcached, imagick, mcrypt, libsmbclient, Zend OPcache

**Nextcloud version:** (see Nextcloud admin page) 20.0.4 - 20.0.4.0

**Where did you install Nextcloud from:** linuxserver/nextcloud:latest

**Signing status:** 
No errors have been found.

**List of activated apps:**
Enabled:
 - accessibility: 1.6.0
 - activity: 2.13.4
 - announcementcenter: 4.0.0
 - apporder: 0.11.0
 - bruteforcesettings: 2.0.1
 - cloud_federation_api: 1.3.0
 - comments: 1.10.0
 - contacts: 3.4.2
 - contactsinteraction: 1.1.0
 - dashboard: 7.0.0
 - dav: 1.16.2
 - deck: 1.2.2
 - federatedfilesharing: 1.10.2
 - federation: 1.10.1
 - files: 1.15.0
 - files_pdfviewer: 2.0.1
 - files_rightclick: 0.17.0
 - files_sharing: 1.12.1
 - files_trashbin: 1.10.1
 - files_versions: 1.13.0
 - files_videoplayer: 1.9.0
 - firstrunwizard: 2.9.0
 - integration_moodle: 0.0.5
 - issuetemplate: 0.7.0
 - logreader: 2.5.0
 - lookup_server_connector: 1.8.0
 - mail: 1.7.2
 - nextcloud_announcements: 1.9.0
 - notifications: 2.8.0
 - oauth2: 1.8.0
 - password_policy: 1.10.1
 - photos: 1.2.1
 - privacy: 1.4.0
 - provisioning_api: 1.10.0
 - recommendations: 0.8.0
 - serverinfo: 1.10.0
 - settings: 1.2.0
 - sharebymail: 1.10.0
 - support: 1.3.0
 - survey_client: 1.8.0
 - systemtags: 1.10.0
 - tasks: 0.13.6
 - text: 3.1.0
 - theming: 1.11.0
 - twofactor_backupcodes: 1.9.0
 - updatenotification: 1.10.0
 - user_status: 1.0.1
 - viewer: 1.4.0
 - weather_status: 1.0.0
 - workflowengine: 2.2.0
Disabled:
 - admin_audit
 - calendar
 - circles
 - encryption
 - files_external
 - groupfolders
 - quota_warning
 - side_menu
 - spreed
 - user_ldap

**Nextcloud configuration:**

```
{
    "defaultapp": "apporder",
    "port": 6379,
    "memcache.local": "\\OC\\Memcache\\APCu",
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "***REMOVED SENSITIVE VALUE***"
    ],
    "dbtype": "mysql",
    "version": "20.0.4.0",
    "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "maintenance": false,
    "app_install_overwrite": [
        "dashboard"
    ],
    "updater.release.channel": "beta",
    "theme": "",
    "loglevel": 2,
    "updater.secret": "***REMOVED SENSITIVE VALUE***"
}
```

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:**
Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4315.7 Safari/537.36

**Operating system:**
Tested on various.  

#### Nextcloud log (data/owncloud.log)
```
Insert your Nextcloud log here
```

#### Browser log
**JS Console:**
```
Uncaught TypeError: OCA.AnnouncementCenter.Comments.CommentsTabView is not a constructor at Object.init (script.js?v=1c7d4f81:42) at HTMLDocument. (script.js?v=1c7d4f81:418) at l (jquery.js:3187) at Object.fireWith [as resolveWith] (jquery.js:3317) at ready (jquery.js:3536) session-heartbeat.js:97 session heartbeat polling started 57globals.js:60 
```

**Comment by Chartman123** [comment]
Possible duplicate of #259?

**Comment by HeyWatchOutDude** [comment]
Same for me. (Nextcloud 20.0.4 - Ubuntu 20.04 LTS)

**Comment by TueTchen** [comment]
Same for me - NC 20.0.4 - CentOS#8

**Comment by dorpauli** [comment]
I got the same problem under Debian 10.7 (NC 20.0.4)

@TueTchen your workaround did not change anything for me. Could you check your fix if it's what you really did?

**Comment by Catmily** [comment]
Can confirm this - workaround didn't work.

NC 20.0.5 - Debian 10

Weirdly, I can't reproduce this in Firefox. This seems to be a Chrome specific issue, at least in my testing.

Nothing weird in log, same error as everyone else. Disabling the comment module didn't fix it. Commenting out the line _did work_, but naturally comments don't work.

**Comment by e-ferrari** [comment]
Same for me.
NC 20.0.3
Ubuntu 20.04
Workaround helps, but poor solution
Bernd

**Comment by kimcdow** [comment]
Same 
NC 20.0.6
Ubuntu 18.04
Refresh works but not a solution

**Comment by nickvergessen** [comment]
please retest with the newest version 5.0.0 in the appstore for Nextcloud 21

**Comment by kcalvelli** [comment]
Verified fixed in 5.0 on Nextcloud 21.  Unfortunately I cannot migrate all of my production servers to 21 just yet, but users have gotten used to the workaround on 20.x.
